dax-prompts/dax-prompts-impl/src/main/java/com/duckduckgo/daxprompts/impl/ReactivateUsersExperiment.kt

```diff
@@ -0,0 +1,103 @@
+ */
+
+package com.duckduckgo.daxprompts.impl
+
+import com.duckduckgo.app.statistics.pixels.Pixel
+import com.duckduckgo.common.utils.DispatcherProvider
+import com.duckduckgo.daxprompts.impl.ReactivateUsersToggles.Cohorts.VARIANT_1
+import com.duckduckgo.daxprompts.impl.ReactivateUsersToggles.Cohorts.VARIANT_2
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.feature.toggles.api.MetricsPixel
+import com.squareup.anvil.annotations.ContributesBinding
+import dagger.SingleInstanceIn
+import javax.inject.Inject
+import kotlinx.coroutines.withContext
+
+interface ReactivateUsersExperiment {
+
+    suspend fun fireDuckPlayerUseIfInExperiment()
+    suspend fun fireSetBrowserAsDefaultIfInExperiment()
+    suspend fun fireDuckPlayerClickIfInExperiment()
+    suspend fun fireChooseYourBrowserClickIfInExperiment()
+    suspend fun fireCloseScreenIfInExperiment()
+    suspend fun fireDismissDuckPlayerIfInExperiment()
+}
+
+@ContributesBinding(
+    scope = AppScope::class,
+    boundType = ReactivateUsersExperiment::class,
+)
+@SingleInstanceIn(AppScope::class)
+class ReactivateUsersExperimentImpl @Inject constructor(
+    private val dispatcherProvider: DispatcherProvider,
+    private val reactivateUsersToggles: ReactivateUsersToggles,
+    private val reactivateUsersPixelsPlugin: ReactivateUsersPixelsPlugin,
+    private val pixel: Pixel,
+) : ReactivateUsersExperiment {
+
+    override suspend fun fireDuckPlayerUseIfInExperiment() {
+        withContext(dispatcherProvider.io()) {
+            if (reactivateUsersToggles.reactivateUsersExperimentMay25().isEnrolledAndEnabled(VARIANT_1)) {
+                reactivateUsersPixelsPlugin.getDuckPlayerUseMetric()?.fire()
+            }
+        }
+    }
+
+    override suspend fun fireSetBrowserAsDefaultIfInExperiment() {
+        withContext(dispatcherProvider.io()) {
+            if (reactivateUsersToggles.reactivateUsersExperimentMay25().isEnrolledAndEnabled(VARIANT_2)) {
+                reactivateUsersPixelsPlugin.getSetBrowserAsDefaultMetric()?.fire()
+            }
+        }
+    }
+
+    override suspend fun fireDuckPlayerClickIfInExperiment() {
+        withContext(dispatcherProvider.io()) {
+            if (reactivateUsersToggles.reactivateUsersExperimentMay25().isEnrolledAndEnabled(VARIANT_1)) {
+                reactivateUsersPixelsPlugin.getDuckPlayerClickMetric()?.fire()
+            }
+        }
+    }
+
+    override suspend fun fireChooseYourBrowserClickIfInExperiment() {
+        withContext(dispatcherProvider.io()) {
+            if (reactivateUsersToggles.reactivateUsersExperimentMay25().isEnrolledAndEnabled(VARIANT_2)) {
+                reactivateUsersPixelsPlugin.getChooseYourBrowserClickMetric()?.fire()
+            }
+        }
+    }
+
+    override suspend fun fireCloseScreenIfInExperiment() {
+        withContext(dispatcherProvider.io()) {
+            if (reactivateUsersToggles.reactivateUsersExperimentMay25().isEnrolledAndEnabled(VARIANT_1)) {
```

Comment:
I don't think you need any checks here. This button exists for both variant 1 and variant 2 and the users can only get to the screen if they are in any of these variants.